subprojects/core/src/main/java/org/gradle/internal/exceptions/ExceptionContextVisitor.java

```diff
@@ -22,4 +22,14 @@ public abstract class ExceptionContextVisitor extends TreeVisitor<Throwable> {
     protected abstract void visitCause(Throwable cause);
 
     protected abstract void visitLocation(String location);
+
+    /**
+     * Should be called after each time this visitor has finished visiting.
+     *
+     * This method can be used to perform any cleanup or final processing related
+     * to the visitor's purpose.
+     */
+    protected void uponFinishedVisiting() {
```

Comment:
🤔 visitEnd()?

Reply:
It's not actually visiting any nodes/anything though.  How about `endVisiting`?